components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -365,12 +365,11 @@ export class GitpodServerImpl<Client extends GitpodClient, Server extends Gitpod
             if (workspace.deleted) {
                 throw new ResponseError(ErrorCodes.PERMISSION_DENIED, "Cannot (re-)start a deleted workspace.");
             }
-
             const envVars = this.userDB.getEnvVars(user.id);
 
             await mayStartPromise;
 
-            return await this.workspaceStarter.startWorkspace({ span }, workspace, user, await envVars);
+            return await this.workspaceStarter.startWorkspace({ span }, workspace, user, await envVars, options);
```

Comment:
here you should explicitly set the options that you pass to `startWorkspace`. Since this is coming from the websocket interface, you don't really know what you're passing along (could be way more than the type, or nothing like the type at all - also see https://www.youtube.com/watch?v=oGeEoaplMWA).

```suggestion
            return await this.workspaceStarter.startWorkspace({ span }, workspace, user, await envVars, {
                forceDefaultImage: !!options.forceDefaultImage
            });
```